india_stocks_api/brokers/angel.py

```
@@ -334,18 +1037,91 @@ def cancel_gtt(self, id: int, instrument: Equity | Future | Option) -> dict:
+                message=resp.get("message")
+            )
+        )
 
-    def get_gtt_list(self, status: list = ["FOR_SETTLEMENT", "CANCELLED", "TRIGGERED"]) -> list:
+    def get_gtt_list(self, status: str) -> GTTRuleResponse:
         jwt_token = context.get_auth_token()
-        payload = {"status": status, "page": 1, "count": 50}
-        resp = get_gtt_list_api(payload, jwt_token)
-        return resp.get('data') or []
+        if not jwt_token:
+            raise RuntimeError("Authentication token is required")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)